### DISCUSSION TITLE
Information on install profiles?

### DISCUSSION THREAD

**Discussion by mrgreen3** [archlinux/archinstall#940] [Q&A]
I can see profiles for inclusion of window managers [i3,deepin etc]. Would like to find out how they are used? Can see how to build iso image but wish to install a WM via installer if possible.

**Comment by Torxed** [comment]
They are sourced during the `select a profile`, if you select `desktop` you will be presented with a second menu asking which desktop, and that’s where the profiles you’re asking for can be selected.

Proframatically all you would do is:
```python
with archinstall.Installer() as installation:
    archinstall.Profile(“gnome”).install()
```

You can generate a configuration with `—dry-run` which you can embed in the ISO.

I suspect that I’ve misunderstood your question or it’s lacking some details. Either way it sounds more like a question/discussion and if that’s the case I’ll go ahead and move this to the discussion section if you don’t mind?

thank you in advance!

**Comment by mrgreen3** [comment]
I have tried dry-run and can see output. If for example I am running this from within a live environment such as arch media can I still use profile to create an install based on it.

More to the point need:

Configure drives, filesystems, locales, mirrors, users and root, base, kernel [etc] i3 [profile drags in xserver and drivers]  to a target device.

Think I could easily modify a profile to suit.

One other thing is a post install script can see an option, would this allow user configs, home directory to be pulled in? 

If I am honest am thinking that archinstall could become a very useful installer for arch based distros. All the elements are there and it works. 

Feel free to move this ;-) am sure I find it.

Many thanks

MrGreen

**Reply by Torxed** [reply]
You can always create your own profile and host it somewhere.
You can have a look at [examples/minimal.py](https://github.com/archlinux/archinstall/blob/master/examples/minimal.py) as a starting point.

And source it with `archinstall https://example.lan/your_profile.py` and it should source it.
That way you can configure your profile to do all the things you wanted, and use that instead of the guided example we ship as a default profile.

I don't see why archinstall couldn't be used by other distros as well. They can tweak and/or create their own scripted installation using the library part only. That's one of the "goals" of the library, to be flexible enough to allow users or groups to create their own spin on the installation process. I also created a [GUI based installer (not working)](https://github.com/Torxed/archinstall_gui) which uses archinstall as an example (severely outdated by now).

Most of our fags etc should support URL loading, so you can use the default arch iso, but source your own installation script, profile, configuration etc remotely.

**Reply by mrgreen3** [reply]
Can see i3.py which I could modify to load Openbox and other needed packages. As list of packages is quite large would it be possible to draw in list from a file [called say packages]? Then need a way to include /etc/skel, could this be done via profile? Finally a way of calling a custom script to do changes needed in new install? I ask as my python skills are limited, can understand basics. Certainly I am more comfortable using bash! Anyway any pointers would be useful as I can then do all the legwork myself.

**Reply by Torxed** [reply]
I'm guessing you would do just fine with simply adding this to your `--conf user_configuration.json`: https://github.com/archlinux/archinstall/blob/16716d94ebf5197f9c4c0e8cbc50709b4e6931ef/examples/custom-command-sample.json#L28-L33

**Reply by mrgreen3** [reply]
I could run a install script for packages and settings from custom-commands. Am assuming those commands are run under chroot? If so my job would be so much easier. Again looking at custom-command I could use my git repo and drag in configs for /etc/skel.

**Reply by Torxed** [reply]
They are run in chroot yes, and they will be executed after the minimal installation is complete :)

**Comment by mrgreen3** [comment]
Looking at guided.py there is a global_menu.run. Am assuming I could run a custom script from here [this will be under arch-chroot]. Not sure yet how to get script onto target drive. Thinking either via curl,wget, git or possibly a package from a custom repo. 

I might use i3 profile for testing as it will install i3, xorg and a log in manager.

Is there a function in archinstall that would allow transfer of a file to target drive from host?

**Reply by Torxed** [reply]
> Is there a function in archinstall that would allow transfer of a file to target drive from host?

We have `with Installation.open('/path/to/file', 'w') as fh:` if that helps? :)
